Content.Server/_Moffstation/GameTicking/Rules/GunGameRuleSystem.cs

```diff
@@ -0,0 +1,229 @@
+    }
+
+    private void OnBeforeSpawn(PlayerBeforeSpawnEvent ev)
+    {
+        var query = EntityQueryEnumerator<GunGameRuleComponent, RespawnTrackerComponent, GameRuleComponent>();
+        while (query.MoveNext(out var uid, out var gunGame, out var tracker, out var rule))
+        {
+            if (!GameTicker.IsGameRuleActive(uid, rule))
+                continue;
+
+            var newMind = _mind.CreateMind(ev.Player.UserId, ev.Profile.Name);
+            _mind.SetUserId(newMind, ev.Player.UserId);
+
+            var mobMaybe = _stationSpawning.SpawnPlayerCharacterOnStation(ev.Station, null, ev.Profile);
+            DebugTools.AssertNotNull(mobMaybe);
+            var mob = mobMaybe!.Value;
+
+            _mind.TransferTo(newMind, mob);
+            _outfitSystem.SetOutfit(mob, gunGame.Gear);
+            EnsureComp<KillTrackerComponent>(mob);
+            EnsureComp<GunGameRewardTrackerComponent>(mob);
+            SpawnCurrentWeapon(ev.Player.UserId, gunGame);
+            _respawn.AddToTracker(ev.Player.UserId, (uid, tracker));
+
+            ev.Handled = true;
+            break;
+        }
+    }
+
+    private void OnSpawnComplete(PlayerSpawnCompleteEvent ev)
+    {
+        EnsureComp<KillTrackerComponent>(ev.Mob);
+        EnsureComp<GunGameRewardTrackerComponent>(ev.Mob);
+        var query = EntityQueryEnumerator<GunGameRuleComponent, RespawnTrackerComponent, GameRuleComponent>();
+        while (query.MoveNext(out var uid, out _, out var tracker, out var rule))
+        {
+            if (!GameTicker.IsGameRuleActive(uid, rule))
+                continue;
+            _respawn.AddToTracker((ev.Mob, null), (uid, tracker));
+        }
+    }
+
+    /// <summary>
+    /// Activates when a kill is reported, ev.Entity corresponds to the person who was killed.
+    /// </summary>
+    private void OnKillReported(ref KillReportedEvent ev)
+    {
+        // Don't want other players picking up somebody's gun
+        DeleteCurrentWeapons(EnsureComp<GunGameRewardTrackerComponent>(ev.Entity));
+
+        var query = EntityQueryEnumerator<GunGameRuleComponent, GameRuleComponent>();
+        while (query.MoveNext(out var uid, out var gungame, out var rule))
+        {
+            if (!GameTicker.IsGameRuleActive(uid, rule))
+                continue;
+
+            // died to something other than a player
+            if (ev.Primary is not KillPlayerSource player)
+            {
+                continue;
+            }
+
+            ProgressPlayerReward(player.PlayerId, gungame);
+            SpawnCurrentWeapon(player.PlayerId, gungame);
+        }
+    }
+
+    /// <summary>
+    /// Appends round end text at the end of the round.
+    /// </summary>
+    protected override void AppendRoundEndText(EntityUid uid, GunGameRuleComponent component, GameRuleComponent gameRule, ref RoundEndTextAppendEvent args)
+    {
+        if (component.Victor != null && _player.TryGetPlayerData(component.Victor.Value, out var data))
```

Comment:
```suggestion
        if (component.Victor is { } victor && _player.TryGetPlayerData(victor, out var data))
```
Totally up to you, but I prefer this sort of nullcheck with var decl to avoid the explicit `.Value` access.